packages/taiko-client/prover/proof_submitter/proof_submitter.go

```diff
@@ -158,6 +161,12 @@ func (s *ProofSubmitter) RequestProof(ctx context.Context, meta metadata.TaikoBl
 				header,
 			)
 			if err != nil {
+				// If request proof has timed out in retry, let's cancel the proof generating and skip
+				if errors.Is(err, proofProducer.ErrProofInProgress) && time.Since(startTime) >= ProofTimeout {
+					log.Error("Request proof has timed out, start to cancel", "blockID", opts.BlockID)
+					_ = s.proofProducer.RequestCancel(ctx, opts)
```

Comment:
No need to check error here?

Reply:
As said before, it's just a one call here, otherwise we're stuck with complicated retry logic.
Or you think there's a strong need here for error and retry process?

Reply:
If so, lets `log.Error` those errors, completely ignore them is not a good practice.

Reply:
OK